src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,80 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+type Props = {
+  todos: Todo[];
+  select: number;
+  setSelect: (select: number) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todos, select, setSelect }) => {
+  const [loading, setLoading] = useState(false);
+  const [user, setUser] = useState<User>();
+
+  const selectedTodo = todos.find(todo => todo.id === select);
+  const foundedUserId = selectedTodo?.userId;
```

Comment:
The `selectedTodo` and `foundedUserId` variables are used to determine the selected todo and its associated user. Ensure that the `select` state is managed consistently across the application to avoid any issues with displaying the correct todo in the modal.